### ISSUE TITLE
Typescript: Cannot Insert object with readonly properties

### ISSUE THREAD

**Issue by Moland9** [edgedb/edgedb-js#282]
When inserting an object in typescript, readonly properties does NOT show up in VS Code autocomplete, and in my case the Nest.js server does not start.

EdgeDB CLI 1.1.0+96c3d69
EdgeDB NPM version: 0.19.8

#
- `default.esdl`:
```
type User {
	required property name -> str {
		constraint exclusive;
	}
	required property identifier -> str {
		readonly := true;
	}
}
```
#
- `user.service.ts`:
```typescript
return e
	.insert(e.User, {
		email,
		identifier,  // This shows error in VS Code.
	})
	.unlessConflict((user) => ({
		on: user.email,
	}));
```
#
- Using REPL works as expected though:
```bash
edgedb> insert User { name := "user1", identifier := "ext-id-user1" };
{default::User {id: be2d9090-9be1-11ec-ac36-a3f5427ffc5a}}

edgedb> select User { id, name, identifier };
{default::User {id: be2d9090-9be1-11ec-ac36-a3f5427ffc5a, name: 'user1', identifier: 'ext-id-user1'}

edgedb> update User filter .name = "user1" set { identifier := "ext-updated-id-user1" };
error: QueryError: cannot update property 'identifier': it is declared as read-only
  ┌─ query:1:56
  │
1 │ update User filter .name = "user1" set { identifier := "ext-updated-id-user1" };
  │                                                        ^^^^^^^^^^^^^^^^^^^^^^ error
```
#
- And the shape is declared in the generated code:
```typescript
export type $UserλShape = $.typeutil.flatten<_std.$Object_7612d407950911ec990f2133715da9d2λShape & {
  "identifier": $.PropertyDesc<_std.$str, $.Cardinality.One, false, false, false>;
  "name": $.PropertyDesc<_std.$str, $.Cardinality.One, false, true, false>;
}>;
```

**Comment by Moland9** [comment]
@Sikarii pointed out that `InsertShape` strips non-writable properties. 
```typescript
// generated/insert.d.ts:25-36
export declare type InsertShape<Root extends ObjectTypeSet> = typeutil.stripNever<
	stripNonWritables<stripBacklinks<Root["__element__"]["__pointers__"]>>
> extends infer Shape
	? Shape extends ObjectTypePointers
		? typeutil.addQuestionMarks<{
				[k in keyof Shape]:
					| pointerToAssignmentExpression<Shape[k]>
					| (pointerIsOptional<Shape[k]> extends true ? undefined | null : never)
					| (Shape[k]["hasDefault"] extends true ? undefined : never);
		  }>
		: never
	: never;
```

I tried commenting out the following and the insert worked as expected
```typescript
export declare type InsertShape<Root extends ObjectTypeSet> = typeutil.stripNever<
	// stripNonWritables<stripBacklinks<Root["__element__"]["__pointers__"]>>
> extends infer Shape
	? Shape extends ObjectTypePointers
		? typeutil.addQuestionMarks<{
				[k in keyof Shape]:
					| pointerToAssignmentExpression<Shape[k]>
					| (pointerIsOptional<Shape[k]> extends true ? undefined | null : never)
					| (Shape[k]["hasDefault"] extends true ? undefined : never);
		  }>
		: never
	: never;
```

**Comment by colinhacks** [comment]
Fixed in `v0.19.10`
